src/sentry/tasks/integrations/github/open_pr_comment.py

```diff
@@ -158,34 +159,47 @@ def safe_for_comment(
                 tags={"type": GithubAPIErrorType.UNKNOWN.value, "code": e.code},
             )
             logger.exception("github.open_pr_comment.unknown_api_error", extra={"error": str(e)})
-        return False
+        return False, []
 
     safe_to_comment = True
-    if pullrequest_resp["state"] != "open":
-        metrics.incr(
-            OPEN_PR_METRICS_BASE.format(key="rejected_comment"), tags={"reason": "incorrect_state"}
-        )
-        safe_to_comment = False
-    if pullrequest_resp["changed_files"] > OPEN_PR_MAX_FILES_CHANGED:
+
+    changed_file_count = 0
+    changed_lines_count = 0
+
+    for file in pr_files:
+        filename = file["filename"]
+        # don't count the file if it was added or is not a Python file
+        if file["status"] == "added" or not filename.endswith(".py"):
+            continue
+
+        changed_file_count += 1
+        changed_lines_count += file["changes"]
+
+    if changed_file_count > OPEN_PR_MAX_FILES_CHANGED:
         metrics.incr(
             OPEN_PR_METRICS_BASE.format(key="rejected_comment"), tags={"reason": "too_many_files"}
         )
         safe_to_comment = False
-    if pullrequest_resp["additions"] + pullrequest_resp["deletions"] > OPEN_PR_MAX_LINES_CHANGED:
+    if changed_lines_count > OPEN_PR_MAX_LINES_CHANGED:
         metrics.incr(
             OPEN_PR_METRICS_BASE.format(key="rejected_comment"), tags={"reason": "too_many_lines"}
         )
         safe_to_comment = False
-    return safe_to_comment
 
+    if not safe_to_comment:
+        pr_files = []
+
+    return safe_to_comment, pr_files
 
-def get_pr_filenames(
-    gh_client: GitHubAppsClient, repository: Repository, pull_request: PullRequest
-) -> List[str]:
-    pr_files = gh_client.get_pullrequest_files(repo=repository.name, pull_number=pull_request.key)
 
+def get_pr_filenames(pr_files: JSONData) -> List[str]:
     # new files will not have sentry issues associated with them
-    pr_filenames: List[str] = [file["filename"] for file in pr_files if file["status"] != "added"]
+    # only fetch Python files
+    pr_filenames: List[str] = [
+        file["filename"]
+        for file in pr_files
+        if file["status"] != "added" and file["filename"].endswith(".py")
```

Comment:
This iteration is very similar to [here](https://github.com/getsentry/sentry/pull/61981/files#diff-ac61a65f5e0d54d5711c6fbf9a39731eb359414ef36809e9938aa8a18a8286e9R169). It might be worth consolidating it in `safe_for_comment`.

Reply:
i moved it out because i want to avoid overloading the `safe_for_comment` function. theoretically you could 1) determine if the function is safe for comment, 2) get the pr filenames, AND 3) get the pr diffs in `safe_for_comment`.

eventually i will also be getting the pr diffs in this function

Reply:
Hmm, I think to avoid the iteration it'd be better to combine 1 and 2 and then fetch the diffs in 3. I think avoiding overloading is good, but step 2 is already being done in 1 so it seems redundant.

Reply:
either way i'd have to have a bit of redundancy in terms of iterating through the files. to get the diffs i'd have to go through the list of PR files from the API response and figure out which ones are not added and are python files again or match by the PR filenames i already filtered by, both of which is are conditional checks

Reply:
or i could just return the json blobs

Reply:
To confirm you're already going through the list of PR files from the response right? Wdym by matching against Pr filenames you already filtered by?

Reply:
i think i can refactor it. i can also early return if we hit too many files or lines early on